src/app/pages/office/cat-b/office.cat-b.page.ts

```diff
@@ -1,58 +1,276 @@
+          dangerousFaultComment.comment,
+        ),
+      );
+
+    } else if (dangerousFaultComment.source === CommentSource.CONTROLLED_STOP) {
+      this.store$.dispatch(AddControlledStopComment(dangerousFaultComment.comment));
+
+    } else if (dangerousFaultComment.source === CommentSource.VEHICLE_CHECKS) {
+      this.store$.dispatch(AddShowMeTellMeComment(dangerousFaultComment.comment));
+    }
   }
 
-  async navigateForward(): Promise<void> {
-    await this.routeByCat.navigateToPage(TestFlowPageNames.REKEY_REASON_PAGE);
+  seriousFaultCommentChanged(seriousFaultComment: FaultSummary) {
+    if (seriousFaultComment.source === CommentSource.SIMPLE) {
+      this.store$.dispatch(
+        AddSeriousFaultComment(seriousFaultComment.competencyIdentifier, seriousFaultComment.comment),
+      );
+    } else if (startsWith(seriousFaultComment.source, CommentSource.MANOEUVRES)) {
+      const segments = seriousFaultComment.source.split('-');
+      const fieldName = segments[1];
+      const controlOrObservation = segments[2];
+      this.store$.dispatch(
+        AddManoeuvreComment(
+          fieldName,
+          CompetencyOutcome.S,
+          controlOrObservation,
+          seriousFaultComment.comment,
+        ),
+      );
+
+    } else if (seriousFaultComment.source === CommentSource.CONTROLLED_STOP) {
+      this.store$.dispatch(AddControlledStopComment(seriousFaultComment.comment));
+    } else if (seriousFaultComment.source === CommentSource.VEHICLE_CHECKS) {
+      this.store$.dispatch(AddShowMeTellMeComment(seriousFaultComment.comment));
+    } else if (seriousFaultComment.source === CommentSource.EYESIGHT_TEST) {
+      this.store$.dispatch(EyesightTestAddComment(seriousFaultComment.comment));
+    }
   }
 
-  async navigateJournal(): Promise<void> {
-    await this.router.navigate([JOURNAL_PAGE]);
+  drivingFaultCommentChanged(drivingFaultComment: FaultSummary) {
+    if (drivingFaultComment.source === CommentSource.SIMPLE) {
+      this.store$.dispatch(
+        AddDrivingFaultComment(drivingFaultComment.competencyIdentifier, drivingFaultComment.comment),
+      );
+    } else if (startsWith(drivingFaultComment.source, CommentSource.MANOEUVRES)) {
+      const segments = drivingFaultComment.source.split('-');
+      const fieldName = segments[1];
+      const controlOrObservation = segments[2];
+      this.store$.dispatch(
+        AddManoeuvreComment(
+          fieldName,
+          CompetencyOutcome.DF,
+          controlOrObservation,
+          drivingFaultComment.comment,
+        ),
+      );
+
+    } else if (drivingFaultComment.source === CommentSource.CONTROLLED_STOP) {
+      this.store$.dispatch(AddControlledStopComment(drivingFaultComment.comment));
+
+    } else if (drivingFaultComment.source === CommentSource.VEHICLE_CHECKS) {
+      this.store$.dispatch(AddShowMeTellMeComment(drivingFaultComment.comment));
+    }
+
   }
 
+  shouldDisplayDrivingFaultComments = (data: CatBUniqueTypes.TestData): boolean => {
```

Comment:
Can we add a TODO to create a new method inside the ```faultCountProvider```, something like getMaximumFaultSumCount. The new method could take in the maximum count i.e. 15 in this case and the category (assume all categories have similar calls so would be nice to centralise)

Reply:
added mes-7281